Zadanie7.3.3/Program.cs

```diff
@@ -0,0 +1,146 @@
+﻿using System;
+using System.Collections.Generic;
+using System.Linq;
+using System.Security.Cryptography.X509Certificates;
+using System.Text;
+using System.Threading.Tasks;
+
+namespace Zadanie7._3._3
+{
+    internal class Program
+    {
+
+        //проверка коррекстности позиций
+        public static void proverka_position(int x, int y)
+        {
+
+            if (x > 8 || x < 1)
+            {
+                Console.WriteLine("Неправильная позиция");
+
+            }
+            else if (y > 8 || y < 1)
+            {
+                Console.WriteLine("Неправильная позиция");
+
+            }
+
+        }
+        //проверка следующего хода ладьи
+        static bool proverka_hoda_ladya(int first_pos, int first_posy,int second_pos, int second_posy, int ferz_x, int ferz_y)
+        {
+            if (first_pos == second_pos||first_posy==second_posy)
+            {
+                Console.WriteLine("Ход возможен");
+                return false;
+            }
+            else if (second_pos==ferz_x&second_posy==ferz_y)
+            {
+                Console.WriteLine("на этом месте стоит другая фигура");
+                return false;
+            }
+            else
+            {
+                Console.WriteLine("Ход невозможен");
+                return true;
+            }   
+        }
+        
+        //проверка сруба
+        public static void next_move_check(int white_x, int white_y,int black_x, int black_y)
+        {
+            if (white_x == black_x & white_y == black_y)
```

Comment:
Используйте ленивые логические операции
```suggestion
            if (white_x == black_x && white_y == black_y)
```